docs/README.md

```diff
@@ -0,0 +1,135 @@
+```yaml
+apiVersion: bmc.tinkerbell.org/v1alpha1
+kind: Job
+metadata:
+  name: job-sample
+spec:
+  machineRef:
+    name: machine-sample
+    namespace: sample
+  tasks:
+    - powerAction: "off"
+    - oneTimeBootDeviceAction:
+        device:
+          - "pxe"
+        efiBoot: false
+    - powerAction: "on"
+```
+The `machineRef` points to the Machine object on the cluster, for which the job is executed. The `tasks` list is a set of ordered actions to be performed on the machine.
+> Note: A single task can only perform one type of action. For example either PowerAction or OneTimeBootDeviceAction.
+### Job Controller
+The job controller watches for Job objects on the cluster. Once a new job object is created, it immediately sets the job condition to `Running` and creates a `Task` object on the cluster for the first item in the tasks list.
+
+The job controller also watches for changes in `Task` objects which have an ownerRef pointing to a Job. Once a Task object status is updated, the job controller checks the conditions on the Task and either marks the Job as Completed/Failed or proceeds to create the next Task object.
+### Task API
+The task type represents a single one-off action performed against a BMC of a physical machine.
+```yaml
+apiVersion: bmc.tinkerbell.org/v1alpha1
+kind: Task
+metadata:
+  name: task-sample
+spec:
+  task:
+    powerAction: "on"
+```
+### Task controller
+The Task controller watches for Task objects on the cluster. When a new Task is created, the controller executes the corresponding action. Once the action is completed, the controller reconciles to check for the state of the physical machine. This ensures the action was completed successdully and marks the `status` as `Completed/Failed` accordingly.
+## Getting Started
+For running Rufio, we require a k8s cluster that has access to the BMC network of the physical machines. 
+
+For the purpose of this tutorial, lets create a [kind](https://kind.sigs.k8s.io/) cluster.
+```bash
+kind create cluster
+```
+Once the cluster is created, we can run the make target to generate a manifest to apply the CRDs and controller deployment.
+```bash
+make release-manifests
+```
+The manifest gets generated under `./out/release/manifest.yaml`, we can then apply the manifest to the cluster.
+```bash
+kubectl apply -f ./out/release/manifest.yaml
+```
+We can now see the Rufio controller pod running on the cluster under the `rufio-system` namespace.
+```bash
+kubectl get po -n rufio-system
+
+NAME                                        READY   STATUS    RESTARTS   AGE
+rufio-controller-manager-6c96d86cf5-kbvdj   1/1     Running   0          5s
+```
+We can now interact with Rufio using its API. Before creating a Mahine object, we need to create a k8s Secret that contains the BMC user credentials for the physical machine. An example Secret,
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bm-auth
+  namespace: sample
+data:
+  username: YWRtaW4=
```

Comment:
i know this is just an example, but maybe this should be something like `root`, etc?

Reply:
Sure! I just grabbed this off k8s docs.